test/jdk/java/awt/image/MemoryLeakTest/MemoryLeakTest.java

```diff
@@ -22,154 +22,117 @@
-  new MemoryLeak().start();
-  Globals.mainThread = Thread.currentThread();
-  try {
-    Thread.sleep(300000);
-  } catch (InterruptedException e) {
-    if (!Globals.testPassed)
-      throw new Exception("MemoryLeakTest failed.");
-  }
-}
-
-}
-
-class TestDialog extends Dialog
-    implements ActionListener {
-
-TextArea output;
-Button passButton;
-Button failButton;
-String name;
-
-public TestDialog(Frame frame, String name)
-{
-  super(frame, name + " Pass/Fail Dialog");
-  this.name = name;
-  output = new TextArea(11, 50);
-  add("North", output);
-  output.append("Do the following steps on Solaris only.\n");
-  output.append("Maximize and minimize the Memory Leak Test window.\n");
-  output.append("Execute the following after minimize.\n");
-  output.append("    ps -al | egrep -i 'java|PPID'\n");
-  output.append("Examine the size of the process under SZ.\n");
-  output.append("Maximize and minimize the Memory Leak Test window again.\n");
-  output.append("Execute the following after minimize.\n");
-  output.append("    ps -al | egrep -i 'java|PPID'\n");
-  output.append("Examine the size of the process under SZ.\n");
-  output.append("If the two SZ values are the same, plus or minus one,\n");
-  output.append("then click Pass, else click Fail.");
-  Panel buttonPanel = new Panel();
-  passButton = new Button("Pass");
-  failButton = new Button("Fail");
-  passButton.addActionListener(this);
-  failButton.addActionListener(this);
-  buttonPanel.add(passButton);
-  buttonPanel.add(failButton);
-  add("South", buttonPanel);
-  pack();
+  static boolean testPassed = false;
+  static Thread mainThread = null;
 }
 
-public void start()
-{
-  show();
-}
-
-public void actionPerformed(ActionEvent event)
-{
-    if ( event.getSource() == passButton ) {
-      Globals.testPassed = true;
-      System.err.println(name + " Passed.");
-    }
-    else if ( event.getSource() == failButton ) {
-      Globals.testPassed = false;
-      System.err.println(name + " Failed.");
+public class MemoryLeakTest {
+    private static final String INSTRUCTIONS =
+        """
+         Do the following steps on Solaris only.
+         Maximize and minimize the Memory Leak Test window.
+         Execute the following after minimize.
+             ps -al | egrep -i 'java|PPID'
+         Examine the size of the process under SZ.
+         Maximize and minimize the Memory Leak Test window again.
+         Execute the following after minimize.
+             ps -al | egrep -i 'java|PPID'
+         Examine the size of the process under SZ.
+         If the two SZ values are the same, plus or minus one,
+         then click Pass, else click Fail.
+        """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame
+            .builder()
+            .title("MemoryLeakTest Instructions")
+            .instructions(INSTRUCTIONS)
+            .rows(15)
+            .columns(40)
+            .testUI(new MemoryLeak())
```

Comment:
It currently creates the MemoryLeak(aka Frame) on the main thread, instead of EDT.

```suggestion
            .testUI(MemoryLeak::new)
```

Reply:
Completely missed this. Thanks!